parachain/pallets/ethereum-beacon-client/src/lib.rs

```diff
@@ -632,11 +647,36 @@ pub mod pallet {
 			Self::deposit_event(Event::BeaconHeaderImported{block_hash: block_root, slot: slot});
 		}
 
-		fn store_execution_header(block_root: H256, header: ExecutionHeaderOf<T>) {
+		fn store_execution_header(block_root: H256, header: ExecutionHeaderOf<T>, slot: u64, beacon_header_hash: H256) {
 			let block_number = header.block_number;
 
 			<ExecutionHeaders<T>>::insert(block_root, header);
 
+			<LatestExecutionHeaderState<T>>::try_mutate(|value| {
+				if block_number > value.block_number {
+					log::info!(
+						target: "ethereum-beacon-client",
+						"💫 Updated latest execution block number to {}.",
+						block_number
+					);
+
+					value.beacon_header_block_root = beacon_header_hash;
+					value.beacon_header_slot = slot;
+					value.block_hash = block_root;
```

Comment:
We're switching between hash and root a bit here. Could we use one of them, eg. root? Not sure if there's a distinction I'm missing here.

```suggestion
					value.beacon_header_block_root = beacon_header_root;
					value.beacon_header_slot = slot;
					value.block_root = block_root;
```

Reply:
I agree it's a little bit confusing! However, this is the terminology used on the consensus layer and execution header respectively, so I am afraid it might more unclear to a reader familiar to Ethereum if I change both to `hash` or `root`.

For illustration purposes:
https://beaconcha.in/slot/4937621
![Screenshot 2022-10-18 at 08 46 30](https://user-images.githubusercontent.com/1788149/196355910-975ede3b-52eb-4c39-ae44-b7e791d2c0fc.png)

https://goerli.etherscan.io/block/7784843
![Screenshot 2022-10-18 at 08 46 56](https://user-images.githubusercontent.com/1788149/196355939-846d22f0-acf2-4da9-96c2-49b16eb5d94b.png)

Even though both values are `H256`, there are subtle differences to how the values came to be. On the beacon chain, `block_root` is the Merkle root of the block, while on the execution chain the block `hash` is just a hash (not sure how they generate those hashes, but it isn't a Merkleization of the header).

An interesting note, on the beacon chain the block root isn't available as a field (besides for a specific API endpoint that returns only the block root), it needs to be calculated.

Reply:
@doubledup agreed about https://github.com/Snowfork/snowbridge/pull/696#discussion_r997036762. Addressed in https://github.com/Snowfork/snowbridge/pull/696#discussion_r997036762, I removed those fields.

Reply:
ah ok, that's the distinction 👌